### PR TITLE
Geometry Analysis and Coverage Dependence Part 1

### PR DESCRIPTION
**Pull request by mjohnson541**
This adds a lot of technology developed for coverage dependence, primarily a set of tools for doing 3D to 2D for adsorbates and transition states.

### CODE REVIEW

**Comment by mjohnson541** [comment]
This also adds SIDT based bond factor estimation.

**Comment by tdprice-858** [comment]
Can we add type hints to function args and returns?  It would also be good to add function descriptions to the code. 
It can be hard to follow without this. And it will be harder in the future if we want to make changes.

**Comment by mjohnson541** [comment]
Deprecated in favor of #52
